contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+     * @param _liquidityToDecrease The amount of liquidity to remove expressed in 18 decimal point
+     */
+    function _removeLiquidity(uint256 _liquidityToDecrease) internal {
+        // unstake the position from the gauge
+        clGauge.withdraw(tokenId);
+
+        (uint128 liquidity,,) = _getPositionInfo();
+        // need to convert to uint256 since intermittent result is to big for uint128 to handle
+        uint128 liqudityToRemove = uint128(uint256(liquidity).mulTruncate(_liquidityToDecrease));
+
+        (uint256 amountWETH, uint256 amountOETHb) = positionManager.decreaseLiquidity(
+            INonfungiblePositionManager.DecreaseLiquidityParams({
+                tokenId: tokenId,
+                liquidity: liqudityToRemove,
+                /**
+                 * Both expected amounts can be 0 since we don't really care if any swaps
+                 * happen just before the liquidity removal.
+                 */
+                amount0Min: 0,
+                amount1Min: 0,
+                deadline: block.timestamp
+            })
+        );
+
+        (uint256 amountWETHCollected, uint256 amountOETHbCollected) = positionManager.collect(
+            INonfungiblePositionManager.CollectParams({
+                tokenId: tokenId,
+                recipient: address(this),
+                amount0Max: type(uint128).max, // defaults to all tokens owed
+                amount1Max: type(uint128).max  // defaults to all tokens owed
+            })
+        );
+
+        updateUnderlyingAssets();
+
+        emit LiquidityRemoved(
+            withdrawLiquidityShare,
+            amountWETH, //removedWETHbAmount
+            amountOETHb, //removedOETHbAmount
+            amountWETHCollected,
+            amountOETHbCollected,
+            underlyingAssets
+        );
+    }
+
+    /**
+     * @dev Perform a swap so that after the swap the ticker has the desired WETH to OETHb token share.
+     */
+    function _swapToDesiredPosition(uint256 _amountToSwap, uint256 _minTokenReceived, bool _swapWETH) internal {
+        IERC20 tokenToSwap = IERC20(_swapWETH ? WETH : OETHb);
+        uint256 balance = tokenToSwap.balanceOf(address(this));
+
+        if(balance < _amountToSwap) {
+            // if swapping OETHb
+            if (!_swapWETH) {
+               uint256 mintForSwap = _amountToSwap - balance;
+               IVault(vaultAddress).mintForStrategy(mintForSwap);
+            } else {
+                revert NotEnoughWethForSwap(balance, _amountToSwap);
+            }
+        }
+
+        // emit an event so it is easier to find correct values off-chain
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionBeforeSwap(WETHPositionBalance, OETHbPositionBalance);
+
+        // Swap it
+        uint256 amountReceived = swapRouter.exactInputSingle(
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: _swapWETH ? WETH : OETHb,
```

Comment:
probably using `tokenToSwap` here would save this conditional check

Reply:
nice agreed: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/306a9b5d536110e61545bfb501797a66ada742e6